azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -518,6 +520,50 @@ private void handleDownloadProject(HttpServletRequest req,
 
   }
 
+    /**
+     * validate readiness of a project and user permission and use
+     * projectManager to purge the project if things looks good
+     **/
+    private void handlePurgeProject(HttpServletRequest req,
+        HttpServletResponse resp, Session session) throws ServletException,
+        IOException {
+        User user = session.getUser();
+        String projectName = getParam(req, "project");
+        HashMap<String, Object> ret = new HashMap<String, Object>();
+        boolean valid = true;
+
+        Project project = projectManager.getProject(projectName);
+        if (project != null) {
+            ret.put("error", "Project " + projectName
+                + " should be deleted before purging");
+            valid = false;
+        }
+
+        // project is already deleted
+        if (valid) {
+            project = projectManager.getInactiveProject(projectName);
+            // only eligible users can purge a project
+            if (!hasPermission(project, user, Type.ADMIN)) {
+                ret.put("error", "Cannot purge. User '" + user.getUserId()
+                    + "' is not an ADMIN.");
+                valid = false;
+                ;
+            }
+        }
+
+        if (valid) {
+            try {
+                projectManager.purgeProject(project, user);
+            } catch (ProjectManagerException e) {
+                ret.put("error", e.getMessage());
+                valid = false;
+            }
+        }
+
+        ret.put("success", valid);
```

Comment:
So the return object can have "success" as key and valid value as false.  Is that OK?